### PR TITLE
A few patches

### PR DESCRIPTION
**Pull request by ailispaw**
I found a few problem with 9p2000.u and 9p2000 during my tests.
I hope it is useful for you.

### CODE REVIEW

File: lib9p/backend/fs.c [hunk 1]
```diff
@@ -395,7 +395,7 @@ fs_create(void *softc, struct l9p_request *req)
 		    mode);
 	}
 
-	if (lchown(newname, file->uid, file->gid) != 0) {
+	if (lchown(newname, file->uid, (gid_t)-1) != 0) {
```
Comment (jceel) [inline lib9p/backend/fs.c]: This is actually masking actual bug - `file->gid` should have been populated in Twalk.

Comment (ailispaw) [inline lib9p/backend/fs.c]: I see.

**Comment by jceel** [comment]
I've merged your changes to lib9p repository: https://github.com/jceel/lib9p/commit/7be3e0b914fbb2d2ecfe2984d02bacfd8e762eae

Please file issues/pull requests against lib9p next time - it's the ultimate "source of truth" that's being used in both bhyve and xhyve. Thank you for contributing!

**Comment by ailispaw** [comment]
Oh, I revised the last commit... will rebase it.

**Comment by ailispaw** [comment]
I have done same as yours and don't need to rebase. Thanks.

**Comment by ailispaw** [comment]
> Please file issues/pull requests against lib9p next time - it's the ultimate "source of truth" that's being used in both bhyve and xhyve.

I see. I will.

**Pull request closed without merging**